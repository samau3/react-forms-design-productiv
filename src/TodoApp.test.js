import React from "react";
import { render, fireEvent } from "@testing-library/react";

import TEST_DATA from "./_testCommon";
import TodoApp from "./TodoApp";

it("renders without crashing", function () {
    render(<TodoApp initialTodos={TEST_DATA} />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<TodoApp initialTodos={TEST_DATA} />);
    expect(asFragment()).toMatchSnapshot();
});

it("creating a new todo", function () {
    const { getByPlaceholderText,
        getByLabelText,
        getAllByRole,
        queryByText,
        container } = render(<TodoApp initialTodos={TEST_DATA} />);

    const titleInput = getByPlaceholderText("Title");
    const descriptionInput = getByPlaceholderText("Description");
    const priorityInput = getByLabelText("Priority:");
    const submitBtn = container.querySelector(".NewTodoForm-addBtn");


    fireEvent.change(titleInput, { target: { value: "get ice cream" } });
    fireEvent.change(descriptionInput, { target: { value: "nom nom nom" } });
    fireEvent.change(priorityInput, { target: { value: 2 } });
    fireEvent.submit(submitBtn);

    const buttons = getAllByRole("button");
    expect(buttons.length).toEqual(9);
    expect(queryByText("get ice cream")).toBeInTheDocument();
    expect(queryByText("nom nom nom")).toBeInTheDocument();
})

it("delete a todo", function () {
    const {
        queryByText,
        container } = render(<TodoApp initialTodos={TEST_DATA} />);


    expect(queryByText("Code!")).toBeInTheDocument();
    expect(queryByText("Write some code")).toBeInTheDocument();

    const deleteBtn = container.querySelector(".EditableTodo-delBtn");
    fireEvent.click(deleteBtn);

    expect(queryByText("Code!")).not.toBeInTheDocument();
    expect(queryByText("Write some code")).not.toBeInTheDocument();
})

it("update a todo", function () {
    const {
        getByPlaceholderText,
        queryByText,
        getByDisplayValue,
        debug,
        container } = render(<TodoApp initialTodos={TEST_DATA} />);

    expect(queryByText("Code!")).toBeInTheDocument();
    expect(queryByText("Write some code")).toBeInTheDocument();

    const editButton = container.querySelector(".EditableTodo-toggle");
    fireEvent.click(editButton);


    const titleInput = container.querySelector("#newTodo-title");
    const descriptionInput = container.querySelector("#newTodo-description");
    // const priorityInput = getByLabelText("Priority:");
    // const deleteBtn = container.querySelector(".EditableTodo-delBtn");

    fireEvent.change(titleInput, { target: { value: "get ice cream" } });
    fireEvent.change(descriptionInput, { target: { value: "nom nom nom" } });

    const submitBtn = container.querySelector("div[id='1'] button.NewTodoForm-addBtn");
    // const submitBtn = firstTodo.querySelector(".EditableTodo-addBtn");
    fireEvent.submit(submitBtn);
    // console.log("submit", submitBtn)
    debug(submitBtn)
    debug()
    expect(queryByText("Code!")).not.toBeInTheDocument();
    expect(queryByText("Write some code")).not.toBeInTheDocument();
    expect(queryByText("get ice cream")).toBeInTheDocument();
    expect(queryByText("nom nom nom")).toBeInTheDocument();
})