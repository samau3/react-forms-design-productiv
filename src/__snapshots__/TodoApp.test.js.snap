// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="TodoApp"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="row"
        >
          <h3>
            Todos
          </h3>
          <div>
            <div>
              <div
                class="EditableTodo"
              >
                <div
                  class="mb-3"
                >
                  <div
                    class="float-right text-sm-right"
                  >
                    <div
                      class="Todo"
                    >
                      <div>
                        <b>
                          Code!
                        </b>
                         
                        <small>
                          (priority: 2)
                        </small>
                      </div>
                      <div>
                        <small>
                          Write some code
                        </small>
                      </div>
                    </div>
                    <button
                      class="EditableTodo-toggle btn-link btn btn-sm"
                    >
                      Edit
                    </button>
                    <button
                      class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                    >
                      Del
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div
                class="EditableTodo"
              >
                <div
                  class="mb-3"
                >
                  <div
                    class="float-right text-sm-right"
                  >
                    <div
                      class="Todo"
                    >
                      <div>
                        <b>
                          Make dinner
                        </b>
                         
                        <small>
                          (priority: 1)
                        </small>
                      </div>
                      <div>
                        <small>
                          Cook something healthy
                        </small>
                      </div>
                    </div>
                    <button
                      class="EditableTodo-toggle btn-link btn btn-sm"
                    >
                      Edit
                    </button>
                    <button
                      class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                    >
                      Del
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div
                class="EditableTodo"
              >
                <div
                  class="mb-3"
                >
                  <div
                    class="float-right text-sm-right"
                  >
                    <div
                      class="Todo"
                    >
                      <div>
                        <b>
                          Go to bed
                        </b>
                         
                        <small>
                          (priority: 3)
                        </small>
                      </div>
                      <div>
                        <small>
                          In bed by 11:15
                        </small>
                      </div>
                    </div>
                    <button
                      class="EditableTodo-toggle btn-link btn btn-sm"
                    >
                      Edit
                    </button>
                    <button
                      class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                    >
                      Del
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      >
        <section
          class="row mb-4"
        >
          <h3>
            Top Todo
          </h3>
          <div
            class="Todo"
          >
            <div>
              <b>
                Make dinner
              </b>
               
              <small>
                (priority: 1)
              </small>
            </div>
            <div>
              <small>
                Cook something healthy
              </small>
            </div>
          </div>
        </section>
        <section
          class="row mb-4"
        >
          <h3
            class="mb-3"
          >
            Add Nü
          </h3>
          <form
            class="NewTodoForm"
          >
            <div
              class="form-group"
            >
              <input
                aria-label="Title"
                class="form-control mb-3"
                id="newTodo-title"
                name="title"
                placeholder="Title"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <textarea
                aria-label="Description"
                class="form-control mb-3"
                id="newTodo-description"
                name="description"
                placeholder="Description"
              />
            </div>
            <div
              class="form-group d-flex justify-content-between"
            >
              <div
                class="w-75 d-flex justify-content-between"
              >
                <label
                  class="d-inline-flex"
                  for="newTodo-priority"
                >
                  Priority:  
                </label>
                <select
                  class="form-control form-control-sm d-inline-flex"
                  id="newTodo-priority"
                  name="priority"
                >
                  <option
                    value="1"
                  >
                    Ultra-Über
                  </option>
                  <option
                    value="2"
                  >
                    Über
                  </option>
                  <option
                    value="3"
                  >
                    Meh
                  </option>
                </select>
              </div>
              <button
                class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
              >
                Gø!
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
